src/cython/vapoursynth.pyx

```diff
@@ -1204,6 +1204,23 @@ cdef FrameProps createFrameProps(RawFrame f):
 # Make sure the FrameProps-Object quacks like a Mapping.
 Mapping.register(FrameProps)
 
+
+@cython.final
+@cython.internal
+cdef class _frame:
+
+    @staticmethod
+    cdef void* getdata(VSFrame* frame, int index, unsigned* flags, const VSAPI* lib) nogil:
+        cdef:
+            unsigned mask = 1 << index+1
+
+        if flags[0] & mask:  # trigger copy-on-write
+            flags[0] &= ~mask  # only do so once, see GH-724
+            return <void*> lib.getWritePtr(frame, index)
+        else:
+            return <void*> lib.getReadPtr(frame, index)
+
+
```

Comment:
Why moving this around, though?  It seems like unnecessary diff to me, and this commit would be the first thing to appear when git-blame'ing it.  Not like it matters much at the end of the day, but still.

Reply:
I was under the assumption it has to be defined before used. apparently that's not necessary, so will revert in the next push.